graphbolt/src/cuda/index_select_impl.cu

```diff
@@ -12,10 +12,41 @@
 #include "../index_select.h"
 #include "./common.h"
 #include "./utils.h"
+#include "cub/cub.cuh"
 
 namespace graphbolt {
 namespace ops {
 
+std::pair<torch::Tensor, torch::Tensor> Sort(
+    torch::Tensor input, int num_bits) {
+  int64_t n_items = input.size(0);
+  auto orig_idx = torch::arange(n_items, input.options().dtype(torch::kLong));
+  auto sorted_array = torch::empty_like(input);
+  auto sorted_idx = torch::empty_like(orig_idx);
+  cuda::CUDAWorkspaceAllocator allocator;
+  AT_DISPATCH_INDEX_TYPES(
+      input.scalar_type(), "SortImpl", ([&] {
+        using IdType = index_t;
+        const auto keys_in = input.data_ptr<index_t>();
+        const int64_t* values_in = orig_idx.data_ptr<int64_t>();
```

Comment:
to confirm: int64_t has the same size as kLong?

Reply:
Yes.